Sources/Models/FaTsaiBrain.swift

```diff
@@ -33,80 +33,98 @@ final class FaTsaiBrain {
     init(totalNumbersCount: Int, numbersRange: (first: Int, last: Int)) {
         self.totalNumbersCount = totalNumbersCount
         self.numbersRange = numbersRange
-    }    
+    }
+    
+    func randomNumbers() -> [Int] {
+        
+        guard requiredNumbersInRange() else {
+            return []
+        }
+        guard numbersInRange(inclusiveNumbers, numbersRange: (numbersRange.first, numbersRange.last)) else {
+            return []
+        }
+        
+        guard numbersInRange(exclusiveNumbers, numbersRange: (numbersRange.first, numbersRange.last)) else {
+            return []
+        }
+        
+        let inclusiveNumbersSet = Set(inclusiveNumbers)
+        let exclusiveNumbersSet = Set(exclusiveNumbers)
+        
+        
+        var result = inclusiveNumbersSet.union(consecutiveNumbers(numbersRange: (first: numbersRange.first, last: numbersRange.last)))
+        
+        guard result.count <= totalNumbersCount else {
+            return []
+        }
+        
+        guard !compareInclusiveNumebr(result, with: exclusiveNumbersSet) else {
+            return []
+        }
+        
+        while result.count < totalNumbersCount {
+            let randomInt = Int.random(in: numbersRange.first...numbersRange.last)
+            if !exclusiveNumbersSet.contains(randomInt) {
+                result.insert(randomInt)
+            }
+        }
+        return Array(result).sorted()
+    }
+}
 
+extension FaTsaiBrain {
+   
     private func requiredNumbersInRange() -> Bool {
         return numbersRange.last - numbersRange.first + 1 >= totalNumbersCount
     }
-
-    private func consecutiveNumbers() -> Set<Int> {
+    
+    
+    private func compareInclusiveNumebr(_ fistList: Set<Int>, with sencondList: Set<Int>) -> Bool {
```

Comment:
I think this the function name is unclear about what inclusive numbers is comparing to. 
It's also returns a `boolean`, so it might be better to name it directly like `isInclusiveNumbersInExclusiveNumbers`. Or maybe something better XDDDD.

Also, do you think we need this check?